### PR TITLE
Inventory

### PR DESCRIPTION
**Pull request by FlxwDNS**

### CODE REVIEW

File: server/src/main/java/net/bytemc/minestom/server/inventory/SingletonInventory.kt [hunk 1]
```diff
@@ -18,7 +19,9 @@ open class SingletonInventory(var title: String, var type: InventoryType, var cl
             if(item is ClickableItem) {
                 item.click(player, clickType.name)
             } else if(item is SwitchItem) {
-                item.click(this, slot)
+                item.click(player,this, slot)
+            } else if(item is ToggleItem) {
```
Comment (HttpMarco) [inline server/src/main/java/net/bytemc/minestom/server/inventory/SingletonInventory.kt:23]: duplicated code -> do it simple

File: server/src/main/java/net/bytemc/minestom/server/inventory/item/impl/switchitem/SwitchEntry.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
+package net.bytemc.minestom.server.inventory.item.impl.switchitem;
+
+import net.bytemc.minestom.server.inventory.item.Item;
+import net.minestom.server.entity.Player;
+
+import java.util.function.Consumer;
+import java.util.function.Predicate;
+
+public final class SwitchEntry {
```
Comment (HttpMarco) [inline server/src/main/java/net/bytemc/minestom/server/inventory/item/impl/switchitem/SwitchEntry.java:9]: Change to privat class

File: server/src/main/java/net/bytemc/minestom/server/inventory/item/impl/switchitem/SwitchItem.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Consumer;
+import java.util.function.Predicate;
+
+@SuppressWarnings("unused")
+public final class SwitchItem implements Item {
+    private final List<SwitchEntry> switches;
+
+    private int currentItem;
```
Comment (HttpMarco) [inline server/src/main/java/net/bytemc/minestom/server/inventory/item/impl/switchitem/SwitchItem.java:18]: direct initialize = 0

File: server/src/main/java/net/bytemc/minestom/server/inventory/item/impl/ToggleItem.java [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    private final Item change;
+
+    private int currentItem;
+    private Predicate<Player> predicate;
+
+    public ToggleItem(ItemStack itemStack, Item change) {
+        this.itemStack = itemStack;
+        this.change = change;
+
+        this.currentItem = 0;
```
Comment (HttpMarco) [inline server/src/main/java/net/bytemc/minestom/server/inventory/item/impl/ToggleItem.java:22]: direct initialize = 0

File: server/src/main/java/net/bytemc/minestom/server/inventory/SingletonInventory.kt (hunk 1 above)
Comment (FlxwDNS) [reply server/src/main/java/net/bytemc/minestom/server/inventory/SingletonInventory.kt:23]: dont work in kotlin

**Pull request merged**